deployments.go

```diff
@@ -77,6 +77,34 @@ func (s *deployerService) Deploy(ctx context.Context, db *gorm.DB, opts DeployOp
 	if err := json.NewEncoder(opts.Output).Encode(&msg); err != nil {
 		return r, err
 	}
-
+	go s.streamEvents(opts)
 	return r, err
 }
+
+func (s *deployerService) streamEvents(opts DeployOpts) {
+	for event := range opts.Events {
+		var msg string
+		if event.Error != nil {
+			msg = fmt.Sprintf("Error: %s", event.Error.Error())
+		} else {
+			if event.Message != "" {
+				msg = fmt.Sprintf("Status: %s", event.Message)
+			}
+		}
+
+		if msg != "" {
+			m := jsonmessage.JSONMessage{Status: msg}
+			if err := json.NewEncoder(opts.Output).Encode(&m); err != nil {
+				close(opts.Events)
```

Comment:
Closing a channel that has outside consumers is potentially dangerous. If the consumer happens to send on this channel, after it's been closed, Go will panic: https://play.golang.org/p/YvdnnAzBSw

If we use a channel based API, it should probably be up to the consumer to close the channel, when it's done producing.